axonserver/src/test/java/io/axoniq/axonserver/component/version/BackwardCompatibilityVersionTest.java

```diff
@@ -0,0 +1,138 @@
+    private final BackwardCompatibilityVersion version4_1 = new BackwardCompatibilityVersion("4.1");
+    private final BackwardCompatibilityVersion version4_1_2 = new BackwardCompatibilityVersion("4.1.2");
+    private final BackwardCompatibilityVersion version4_1_3 = new BackwardCompatibilityVersion("4.1.3");
+    private final BackwardCompatibilityVersion version4_1_4 = new BackwardCompatibilityVersion("4.1.4");
+    private final BackwardCompatibilityVersion version4_2 = new BackwardCompatibilityVersion("4.2");
+    private final BackwardCompatibilityVersion version4_2_1 = new BackwardCompatibilityVersion("4.2.1");
+    private final BackwardCompatibilityVersion version5 = new BackwardCompatibilityVersion("5");
+    private final BackwardCompatibilityVersion version5_1 = new BackwardCompatibilityVersion("5.1");
+    private final BackwardCompatibilityVersion version5_1_3 = new BackwardCompatibilityVersion("5.1.3");
+
+    @Test
+    public void testMatch() {
+        boolean match = version4_1_3.match(new BackwardCompatibilityVersion("4.1.3"));
+        assertTrue(match);
+
+        match = version4_1_3.match(new BackwardCompatibilityVersion("4.1"));
+        assertFalse(match);
+    }
+
+    @Test
+    public void testGreaterOrEqualThan() {
+        boolean greaterOrEqual = version4_1.greaterOrEqualThan(version4_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_1_2);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_1_4);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_2_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_0_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_1);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_2);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_0);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5);
+        assertFalse(greaterOrEqual);
+    }
+
+    @Test
+    public void testGreaterOrEqualThanWithPatch() {
```

Comment:
You can introduce jUnitParams test https://github.com/Pragmatists/JUnitParams in order to ease the readability and maintenance.